libraries/AP_Scripting/applets/switch_rates.lua

```diff
@@ -0,0 +1,176 @@
+--[[
+   rate switch utility. This helps to compare different tunes in-flight to compare performance
+--]]
+
+---@diagnostic disable: param-type-mismatch
+
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+local PARAM_TABLE_KEY = 33
+local PARAM_TABLE_PREFIX = "RTSW_"
+local PARAM_BACKUP_TABLE_KEY = 34
+local PARAM_BACKUP_TABLE_PREFIX = "X_"
+
+local UPDATE_RATE_HZ = 4
+
+-- bind a parameter to a variable, old syntax to support older firmware
+function bind_param(name)
+   local p = Parameter()
+   if not p:init(name) then
+      return nil
+   end
+   return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   local p = bind_param(PARAM_TABLE_PREFIX .. name)
+   assert(p, string.format("could not find parameter %s", name))
+   return p
+end
+
+-- add a backup parameter with appropriate length and bind it to a variable
+function bind_add_backup_param(name, idx, default_value)
+   -- shorten pname to fit with X_ prefix
+   local short_name = string.sub(name, math.max(1, 1 + string.len(name) - (16 - string.len(PARAM_BACKUP_TABLE_PREFIX))))
+   assert(param:add_param(PARAM_BACKUP_TABLE_KEY, idx, short_name, default_value), string.format('could not add param %s', short_name))
+   local p = bind_param(PARAM_BACKUP_TABLE_PREFIX .. short_name)
```

Comment:
```suggestion
   local p = Parameter(PARAM_BACKUP_TABLE_PREFIX .. short_name)
```
I suspect this will fix the warning. You can remove the `bind_param` helper.